app/models/audit/event/fetch.rb

```diff
@@ -15,6 +16,7 @@ def initialize(
         @user = user
         @client_ip = client_ip
         @resource = resource
+        @resource_id = resource_id
```

Comment:
Could you add a note to explain why we're sometimes getting the `resource_id` directly from a variable of that name, and sometimes from `@resource.id`?  Depending on the answer to that, I would consider only allowing `resource_id`, and letting clients of this class who hold a `resource` be responsible for passing `resource.id`.

Separately, given the current design, I would create a _method_ `resource_id` which has looks like:
```
@resource_id ||= @resource_id_ || @resource.id
```
or similar.

Reply:
The use ```resource_id``` is for specific case when the resource does not exist. In that case calling ```resource.id``` will raise an error. We need that resource's id for audits and logs purposes. 
 
And about that Second note, it sounds right, I'll check if that breaks anything (even though it shouldn't)

Reply:
> We need that resource's id for audits and logs purposes.

If it doesn't exist, how are you populating `resource_id`?

Reply:
Ignore last question, I see what you are doing after looking at the rest of the code.

The real problem here is sort of legacy one: mixing the Command object and code from the rails "concern".  It would be cleaner if the database lookup were being done by the Command object -- it would also make the whole thing more testable too.

If you don't want to make that change (it's not huge, but not trivial either), I would suggest that you rename `resource_id` to `requested_resource_id` and add a note that `@resource` may be nil sometimes, but `requested_resource_id` will never be.

Reply:
After the last commit there is only use of resource_id, which seems to be by the guidelines you suggest

Reply:
Much better.  Thanks!